packages/nodes-base/nodes/MQTT/GenericFunctions.ts

```diff
@@ -49,5 +49,15 @@ export const createClient = async (credentials: MqttCredential): Promise<MqttCli
 		clientOptions.rejectUnauthorized = credentials.rejectUnauthorized;
 	}
 
-	return await connectAsync(clientOptions);
+	return await new Promise((resolve, reject) => {
+		const client = connect(clientOptions);
+
+		client.on('connect', () => {
```

Comment:
we should use `once` instead of `on`, and remove both the listeners when this promise settles. That's a pattern we should follow for all nodes,to avoid reconnection events trying to trigger these even listeners.